packages/openapi3/src/openapi.ts

```diff
@@ -1824,3 +1827,13 @@ function sortOpenAPIDocument(doc: OpenAPI3Document): void {
     doc.components.parameters = sortObjectByKeys(doc.components.parameters);
   }
 }
+
+async function loadXmlModule(sourceKeys: MapIterator<string>) {
```

Comment:
I think I like the idea of having a module system like that you can pass, however I think need to change a few things:
1. Need some type, cannot pass this xml module as any all over.
2. The way it is done is not reliable. It should just try to import the module. They might not use anything from the xml library and not import it but still might need to add annotation as the default in typespec are different.
  - For that you need to `try catch await` the import of the `xml` library

I think what you can do is something like that

```
export function resolveXmlModule(): XmlModule | undefined {
   const xml = await tryImport("@typespec/xml");
   if(xml === undefined) return undefined;

   return {
     addXmlMetadata: (...) => {

     }
   }
}
```

alternatively you could also just get rid of the module system and just alaways call `addXmlMetadata` and that deals with figuring out if xml is there or not